### ISSUE TITLE
Copying the file to running NAV Container on Windows 10 fails with "filesystem operations against a running Hyper-V container are not supported"

### ISSUE THREAD

**Issue by ichladil** [Microsoft/navcontainerhelper#49]
Environment: Windows 10 and navcontainerhelper 0.2.1.3
Action: Running Import-ObjectsToNavContainer -containerName MyContainer -objectsFile "C:\WorkingFolder\MyFOB.fob" 

Error: _Error response from daemon: filesystem operations against a running Hyper-V container are not supported_

This seems to be caused by the issue described e.g. here:  https://stackoverflow.com/questions/45654570/unable-to-copy-to-windows-container-docker-cp-in-windows-10 

The suggested workaround for running container on Windows 10 might be to change the  Copy-FileToNavContainer command to make it two steps:
1. Use one of the mapped folders and copy the file(s) to the mapped folder
2. Enter PS Session and copy it from the mapped folder to the destination directory

**Comment by Koubek** [comment]
Yes, that's actually true. You can't copy files into the containers being running.

Of course, your proposal is valid and very logical. I have another idea but the my one is really a very alternative one (I can't even test it right now as I am on vacation and don't have my laptop with me just now). It probably could work for small files but for a big FOB...??? The idea is pass the file *content* using `docker exec` (`'fob content' | Set-Content target_filename`).

**Comment by Koubek** [comment]
Or maybe another option (again, not tested):
https://blogs.msdn.microsoft.com/luisdem/2016/08/31/powershell-how-to-copy-a-local-file-to-remote-machines/

**Comment by freddydk** [comment]
Thanks, I will probably remove Copy-FileToNavContainer and Copy-FileFromNavContainer and then handle this in the 3 places where they are used. In these situations, I have all shared folders and can just use one of these very very easily (instead of using c:\run in the container).
Didn't think of it as the copy functions works fine on Windows Server 2016.

**Comment by Koubek** [comment]
BTW - This one works like a charm (tested on Win10, WinServer 2016):
```PowerShell
$ContSession = New-PSSession -ContainerId $cid -RunAsAdministrator
Copy-Item -Path [your_local_file] -Destination [in_container_destination_path] -ToSession $ContSession
```
The solution doesn't dependent on a mapped folder (docker volumes).

**Comment by freddydk** [comment]
Basically what I will do is to replace these lines:
        $containerAppFile = Join-Path "c:\run" ([System.IO.Path]::GetFileName($appFile))
        Copy-FileToNavContainer -containerName $containerName -localPath $appFile -containerPath $containerAppFile
        $copied = $true
with:
        $sharedFolders = Get-NavContainerSharedFolders -containerName $containerName
        if ($sharedFolders.Count -eq 0) {
            throw "Cannot copy objects file to container when no folders are shared"
        }
        $destFolder = $sharedFolders.Keys.GetEnumerator() | Select-Object -First 1
        Copy-Item -path $objectsFile -Destination $destFolder
        $containerObjectsFile = Join-Path ($sharedFolders.Values.GetEnumerator() | Select-Object -First 1) ([System.IO.Path]::GetFileName($objectsFile))
        $copied = $true

the copied file will be deleted with copied = true.

**Comment by freddydk** [comment]
Wow - didn't see Jakub's suggestion before positing this - I will definitely use that.

**Comment by freddydk** [comment]
$id = Get-NavContainerId -containerName $containerName 

        $session = New-PSSession -ContainerId $id -RunAsAdministrator -ErrorAction Ignore
        if ($session) {
            Copy-Item -Path $localPath -Destination $containerPath -ToSession $session
            Remove-PSSession -Session $session
        } else {
            docker cp $localPath ${id}:$containerPath
        }

**Comment by freddydk** [comment]
Checked in, will be included in the next gallery update

**Comment by freddydk** [comment]
Fixed in 0.2.2.0
https://www.powershellgallery.com/packages/navcontainerhelper/0.2.2.0
